public/pages/CreateMonitor/components/MonitorExpressions/expressions/WherePopover.js

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { Component } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiText,
+  EuiBadge,
+} from '@elastic/eui';
+import _ from 'lodash';
+import { POPOVER_STYLE, EXPRESSION_STYLE, WHERE_BOOLEAN_FILTERS } from './utils/constants';
+import {
+  getOperators,
+  displayText,
+  validateRange,
+  isNullOperator,
+  isRangeOperator,
+  validateWhereFilter,
+} from './utils/whereHelpers';
+import { hasError, isInvalid } from '../../../../../utils/validate';
+import {
+  FormikComboBox,
+  FormikSelect,
+  FormikFieldNumber,
+  FormikFieldText,
+} from '../../../../../components/FormControls';
+import { getFilteredIndexFields, getIndexFields } from './utils/dataTypes';
+import { TRIGGER_COMPARISON_OPERATORS } from '../../../../CreateTrigger/containers/DefineBucketLevelTrigger/DefineBucketLevelTrigger';
+import { DATA_TYPES } from '../../../../../utils/constants';
+
+export const ALLOWED_TYPES = ['number', 'text', 'keyword', 'boolean'];
+
+export default class WherePopover extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  componentDidMount() {
+    this.setState({ isOpen: this.props.openedState });
+  }
```

Comment:
We don't need to set this state after mounting, it is fine to do it in the constructor

Reply:
Good point. Revised.